src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -600,4 +607,22 @@ private static function initFormats()
             'debug_nomax' => ' %current% [%bar%] %elapsed:6s% %memory:6s%',
         );
     }
+
+    /**
+     * @param string $line
+     * @return bool
+     */
+    private function adjustBarWidthToWindowWidth($line)
+    {
+        $lineLength = Helper::strlenWithoutDecoration($this->output->getFormatter(), $line);
+        $windowWidth = exec('tput cols');
```

Comment:
This is not enough... what about windows? Look at https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Console/Application.php#L734